Automattic/jetpack
# AI Extension: improve performance bug when extending blocks with AI Assistant

These changes solve a performance problem caused by poor hooks implementation when extending components with the AI extension.
* Removes the useTextContentFromSelectedBlocks() custom hook
* Refact -> move code to the `<AiAssistantDropdown />`
* Clean

Fixes #

## Proposed changes:

* AI Extension: improve performance bug when extending blocks with AI Assistant

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:







**Test on Simple, Atomic, and Jetpack sites**

Test the AI Extension:

* Confirm it extends the `core/paragraph`, `core/heading`,  and `core/list` blocks. Visually, the block toolbar shows the AI button.

<img width="646" alt="Screenshot 2023-10-19 at 17 42 03" src="https://github.com/Automattic/jetpack/assets/77539/ec73f617-3a72-4583-b5de-6aa91602d7ca">

* Test all AI Extension features, such as `translate`, `extended`, etc.
* Confirm the suggestion request/response.
  * Open the network tab of your dev tool
  * Confirm there is only one suggestion request. Take a look at the payload

<img width="783" alt="image" src="https://github.com/Automattic/jetpack/assets/77539/5a344166-87ae-42d1-b21b-64bb938d2b84">


* You may like to take a look at the debug() lines
  * Confirm the app registers the events
  * Confirm the AI suggestion request/response logs 
<img width="780" alt="image" src="https://github.com/Automattic/jetpack/assets/77539/5a0171c3-e675-45a7-833a-0c1930e5be3c">

_**In terms of performance, test with a super long post to perceive the delay when you type text in the editor.**_
